microsoft/testsuites/kselftest/kselftest.py

```diff
@@ -138,30 +162,44 @@ def _install(self) -> bool:
                 sudo=True,
             )
 
-            # build kselftests
             self.node.tools[Make].run(
-                "KBUILD_OUTPUT=build -C tools/testing/selftests gen_tar",
+                "headers",
                 cwd=kernel_path,
                 sudo=True,
                 expected_exit_code=0,
-                expected_exit_code_failure_message="could not generate kselftest tar.",
+                expected_exit_code_failure_message="failed to build kernel headers.",
             ).assert_exit_code()
 
-        tool_path = self.get_tool_path(use_global=True)
-        self.node.tools[Tar].extract(
-            str(self._remote_tar_path), str(tool_path), sudo=True
-        )
-        self._log.debug(f"Extracted tar from path {self._remote_tar_path}!")
+            # build and install kselftests
+            self.node.execute(
+                cmd=f"./kselftest_install.sh {self._kself_installed_dir}",
+                shell=True,
+                cwd=PurePosixPath(kernel_path, "tools/testing/selftests"),
+                sudo=True,
+                expected_exit_code=0,
+                expected_exit_code_failure_message="fail to build & install kselftest",
+            ).assert_exit_code()
+            # change permissions of kselftest-packages directory
+            # to run test as non root user.
+            self.node.execute(
+                f"chmod -R 777 {self._kself_installed_dir}", sudo=True, shell=True
+            )
 
         return self._check_exists()
 
     def run_all(
-        self, test_result: TestResult, environment: Environment, log_path: str
+        self,
+        test_result: TestResult,
+        environment: Environment,
+        log_path: str,
+        run_test_as_root: bool = False,
     ) -> List[KselftestResult]:
+        # Executing kselftest as root may cause
+        # VM to hang
```

Comment:
why?

Reply:
I had a specific case where I observed this, I did not save the reference. Let me try to find it.

My understanding is that without sudo that test does not run.
![image](https://github.com/microsoft/lisa/assets/26188281/f71dfa3f-213b-4c65-84f4-09092fc06433)

Reply:
https://www.spinics.net/lists/linux-kselftest/msg15551.html